pypeit/wavetilts.py

```diff
@@ -188,34 +188,45 @@ def show(self, waveimg=None, wcs_match=True, in_ginga=True, show_traces=False):
                 If True, show the traces of the tilts on the image.
 
         """
+        # get tilt_img_dict
+        if (Path(self.calib_dir).resolve() / self.tiltimg_filename).exists():
+            tilt_img_dict = buildimage.TiltImage.from_file(Path(self.calib_dir).resolve() / self.tiltimg_filename)
+        else:
+            msgs.error(f'Tilt image {str((Path(self.calib_dir).resolve() / self.tiltimg_filename))} NOT FOUND.')
+
         # get slits
+        slitmask = None
         if (Path(self.calib_dir).resolve() / self.slits_filename).exists():
             slits = slittrace.SlitTraceSet.from_file(Path(self.calib_dir).resolve() / self.slits_filename)
-            slitmask = slits.slit_img(initial=True, flexure=self.spat_flexure)
-            left, right, mask = slits.select_edges(flexure=self.spat_flexure)
-            gpm = mask == 0
+            _slitmask = slits.slit_img(initial=True, flexure=self.spat_flexure)
+            if tilt_img_dict.image.shape != _slitmask.shape:
+                msgs.warn('Tilt image and other calibration frames have different shapes. '
```

Comment:
I don't think this is correct. Why are you throwing a warning here? The tilt_img shape can differ from the _slitmask shape for instruments where that is often the case (i.e. X-shooter where they don't take binned arcs), and it should not trigger a warning because nothing is going wrong. Furthermore, the resizing routine will just return the _slitmask if the shapes are the same so the if else is not needed.

Reply:
@jhennawi the warning here is about the slit traces and the wavelength image. arc resize routine allow me to resize  `_slitmask` but not `slits` (and therefore `waveimg`). This is why there is an if/else. If the sizes are different I don't want to use `slits` (because I cannot resize it). And if I don't use `slits` I cannot create the `waveimg`. Hence the warning, I cannot load those 2 things. I can remove the part where it mention the difference in shapes, but I prefer to leave the part where I warn that slits and waveimg cannot be loaded.

Reply:
here is a resizing routine for the slits as well. Again, you should not be throwing any warnings here because the slits being a different shape than the arc/tilt image is a common occurrence for X-shooter.  See the __init__.py method in wavetilts.py. There is a method called arc.resize_slits2arc.  

What you should be doing here is getting rid of the if/else because the routines are designed to work no matter what the shapes are as long as they are integer multiples of each other. 

_slitmask = slits.slit_img(initial=True, flexure=self.spat_flexure)
slitmask = arc.resize_mask2arc(tilt_img_dict.image.shape, _slitmask)
_left, _right, _mask = slits.select_edges(flexure=self.spat_flexure)
left = arc.resize_slits2arc(tilt_img_dict.image.shape, _slitmask.shape, _left)
right = left = arc.resize_slits2arc(tilt_img_dict.image.shape, _slitmask.shape, _right)

Then I'm a bit unclear on what you are doing with the mask, but you can I think use the arc.resize_mask2arc to reshape that as well. No if/then is needed and no warnings should be thrown. The resizing routines return the input image if the shapes are the same.

Reply:
Done.